packages/block-editor/src/components/block-list/insertion-point.js

```diff
@@ -94,13 +95,24 @@ function InsertionPointInserter( {
 
 function InsertionPointPopover( {
 	clientId,
+	rootClientId,
 	isInserterShown,
 	isInserterForced,
 	setIsInserterForced,
 	containerRef,
 	showInsertionPoint,
 } ) {
-	const element = getBlockDOMNode( clientId );
+	const appenderMap = useContext( AppenderContext );
+	const element = useMemo( () => {
+		if ( clientId ) {
+			return getBlockDOMNode( clientId );
+		}
+
+		// Can't find the element, might be at the end of the block list, or inside an empty block list.
+		// We instead try to find the "Appender" and place the indicator above it.
+		// `rootClientId` could be null or undefined when there's no parent block, we normalize it to an empty string.
+		return appenderMap.get( rootClientId || '' );
```

Comment:
Thinking of some different approaches,`getBlockDOMNode` just does a `getElementById` call, could it be an option to do something similar for the appender using `querySelector`? e.g 
```js
document.querySelector( `#block-${ rootClientId } .block-list-appender` );
```

Reply:
That's actually my first attempt. However, I just don't feel like it's "react"ive enough as we'll be relying on the DOM structure and classnames.

Reply:
I think we should go for the simpler solution. We're doing the same for block nodes, and this should be revised together if we'd like to avoid querying the DOM.

Reply:
It's not as _simple_ as just a `querySelector` call though. Since `clientId` could be `null`, also could `rootClientId`. We have to add some more code to be able to select the parent of the appender.